src/app/services/userdata.service.ts

```diff
@@ -0,0 +1,86 @@
+// Angular imports
+import { Injectable } from '@angular/core';
+import { Router } from '@angular/router';
+import { HttpClient } from '@angular/common/http';
+import { catchError, Observable } from 'rxjs';
+
+// Interfaces
+import { User } from '../interfaces/User';
+import { LoginData } from '../interfaces/loginData';
+import { RegistrationData } from '../interfaces/registrationData';
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export class UserdataService
+{
+  // Mockusers
+  // Will use API to get user info later on
+  private readonly mockUser: User =
+  { 
+    id: 0,
+    name: "John Doe",
+    mail: "johndoe@mail.com", 
+    password: "test123",
+    phone: "0123/45 67 89",
+    type: "private",
+    city: "Mechelen",
+    zipcode: 2800,
+    street: "Koningin Astridlaan",
+    house: "10"
+
+  }
+
+  // API connection
+  readonly ROOT_URL = 'http://localhost:6060';
+  customers!: Observable<any>;
+
+  // Variables
+  authenticated = false;
```

Comment:
Noticed an issue with this. Variables aren't kept between page reloads. This will need to be changed to checking the browsers session data.
In my auth branch I slightly refactored your code, I noticed that after login redirect it works fine, but reloading the page resets the variables, causing pages to think the user isn't logged in.

Reply:
I read something about local storage, would that be a solution or will it just create more problems?